File: frontend/lib/models/student.dart
class Student {
  final String rollNo;
  final String photoUrl;
  final String fullName;
  final String gender;
  final String email;
  final String branch;
  final String section;
  final String course;
  final String fatherName;
  final String motherName;
  final String dateOfBirth;
  final String admissionDate;
  final String casteCategory;
  final String aadharNo;
  final String parentMobile;
  final String studentMobile;
  final String ssc;
  final String inter;
  final String courseCompletionYear;

  Student({
    required this.rollNo,
    required this.photoUrl,
    required this.fullName,
    required this.gender,
    required this.email,
    required this.branch,
    required this.section,
    required this.course,
    required this.fatherName,
    required this.motherName,
    required this.dateOfBirth,
    required this.admissionDate,
    required this.casteCategory,
    required this.aadharNo,
    required this.parentMobile,
    required this.studentMobile,
    required this.ssc,
    required this.inter,
    required this.courseCompletionYear,
  });

  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
      rollNo:
          json['rollNo']?.toString() ??
          json['Roll_No']?.toString() ??
          json['roll_no']?.toString() ??
          '',
      photoUrl:
          json['photoUrl']?.toString() ??
          json['Photo_URL']?.toString() ??
          json['photo_url']?.toString() ??
          '',
      fullName:
          json['fullName']?.toString() ??
          json['Full_Name']?.toString() ??
          json['full_name']?.toString() ??
          '',
      gender: json['gender']?.toString() ?? '',
      email: json['email']?.toString() ?? '',
      branch: json['branch']?.toString() ?? '',
      section: json['section']?.toString() ?? '',
      course: json['course']?.toString() ?? '',
      fatherName:
          json['fatherName']?.toString() ??
          json['father_name']?.toString() ??
          '',
      motherName:
          json['motherName']?.toString() ??
          json['mother_name']?.toString() ??
          '',
      dateOfBirth:
          json['dateOfBirth']?.toString() ??
          json['date_of_birth']?.toString() ??
          '',
      admissionDate:
          json['admissionDate']?.toString() ??
          json['admission_date']?.toString() ??
          '',
      casteCategory:
          json['casteCategory']?.toString() ??
          json['caste_category']?.toString() ??
          '',
      aadharNo:
          json['aadharNo']?.toString() ?? json['aadhar_no']?.toString() ?? '',
      parentMobile:
          json['parentMobile']?.toString() ??
          json['parent_mobile']?.toString() ??
          '',
      studentMobile:
          json['studentMobile']?.toString() ??
          json['student_mobile']?.toString() ??
          '',
      ssc: json['ssc']?.toString() ?? '',
      inter: json['inter']?.toString() ?? '',
      courseCompletionYear:
          json['courseCompletionYear']?.toString() ??
          json['course_completion_year']?.toString() ??
          '',
    );
  }
}

File: backend/src/main/java/com/nl2sql/backend/entity/Student.java
package com.nl2sql.backend.entity;

import java.time.LocalDateTime;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "students")
@Data
public class Student {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "Roll_No", unique = true, nullable = false)
    private String rollNo;

    @Column(name = "Full_Name")
    private String fullName;

    @Column(name = "Photo_URL", columnDefinition = "TEXT")
    private String photoUrl;

    @Column(name = "Date_of_Birth")
    private String dateOfBirth;

    @Column(name = "Gender")
    private String gender;

    @Column(name = "Father_Name")
    private String fatherName;

    @Column(name = "Mother_Name")
    private String motherName;

    @Column(name = "Caste_Category")
    private String casteCategory;

    @Column(name = "Admission_Date")
    private String admissionDate;

    @Column(name = "Course_Completion_Year")
    private Integer courseCompletionYear;

    @Column(name = "Parent_Mobile")
    private Long parentMobile;

    @Column(name = "Student_Mobile")
    private Long studentMobile;

    @Column(name = "Email")
    private String email;

    @Column(name = "Course")
    private String course;

    @Column(name = "Branch")
    private String branch;

    @Column(name = "Section")
    private String section;

    @Column(name = "Aadhar_No")
    private String aadharNo;

    @Column(name = "SSC")
    private Double ssc;

    @Column(name = "Inter")
    private Double inter;

    @Column(name = "created_at", insertable = false, updatable = false)
    private LocalDateTime createdAt;
}
